src/main/scala/org/constellation/infrastructure/endpoints/CheckpointEndpoints.scala

```diff
@@ -57,14 +58,16 @@ class CheckpointEndpoints[F[_]](implicit F: Concurrent[F], C: ContextShift[F]) e
   private def handleFinishedCheckpointEndpoint(
     metrics: Metrics,
     snapshotService: SnapshotService[F],
-    checkpointAcceptanceService: CheckpointAcceptanceService[F]
+    checkpointAcceptanceService: CheckpointAcceptanceService[F],
+    tipService: ConcurrentTipService[F]
   ): HttpRoutes[F] = HttpRoutes.of[F] {
     case req @ POST -> Root / "finished" / "checkpoint" =>
       for {
         finishedCheckpoint <- req.decodeJson[FinishedCheckpoint]
         baseHash = finishedCheckpoint.checkpointCacheData.checkpointBlock.baseHash
         _ <- logger.debug(s"Handle finished checkpoint for cb: $baseHash")
         _ <- metrics.incrementMetricAsync[F]("peerApiRXFinishedCheckpoint")
+        _ <- tipService.registerUsages(finishedCheckpoint)
```

Comment:
I guess it's not harmful to register usages here and in accept method, as we are keeping child hashes as a set right?

Reply:
exactly, that's the point. we need to register usages here as this is the first entry point to the node. registering in acceptance is a backup